src/main/java/seedu/address/ui/PersonCard.java

```diff
@@ -69,16 +68,16 @@ public PersonCard(Person person, int displayedIndex) {
         phone.setText(person.getPhone().value);
         address.setText(person.getAddress().value);
         email.setText(person.getEmail().value);
-        person.getMedCons().stream()
-                .sorted(Comparator.comparing(MedCon::toString))
-                .forEach(medCon -> medCons.getChildren().add(new Label(medCon.toString())));
         priority.setText(person.getPriority().priority);
-        person.getTags().stream()
-                .sorted(Comparator.comparing(tag -> tag.tagName))
-                .forEach(tag -> tags.getChildren().add(new Label(tag.tagName)));
-        person.getAppointments()
-                .stream()
-                .sorted(Comparator.comparing(Appointment::getAppointmentStartDateTime))
-                .forEach(appointment -> appointments.getChildren().add(new Label(appointment.toString())));
+        addLabelsToFlowPane(person.getMedCons(), medCons);
+        addLabelsToFlowPane(person.getTags(), tags);
+        addLabelsToFlowPane(person.getAppointments(), appointments);
     }
+
+    private <T extends Comparable<T>> void addLabelsToFlowPane(Set<T> items, FlowPane flowPane) {
+        items.stream()
+                .sorted(Comparator.naturalOrder())
+                .forEach(item -> flowPane.getChildren().add(new Label(item.toString())));
+    }
```

Comment:
have you smoke-tested it? does everything show up when running the actual prog?

Reply:
Yep the sample data loads up